radar/radar.go

```diff
@@ -68,7 +68,17 @@ func NewRadar(
 
 func (radar *Radar) Scanner(logger lager.Logger, resourceName string) ifrit.Runner {
 	return ifrit.RunFunc(func(signals <-chan os.Signal, ready chan<- struct{}) error {
-		ticker := time.NewTicker(radar.interval)
+		var ticker *time.Ticker
+		resourceConfig, err := radar.getResourceConfig(logger, resourceName)
```

Comment:
not too worried about those lower-level details, just that mutating the object is riskier (now have to potentially deal with concurrent calls to scan, though in this case they're wrapped in a db lock). imo it's easier to keep track of return values vs. who's changing what state and having two implicitly inter-dependent code paths.